template/base/miner.py

```diff
@@ -170,33 +106,23 @@ def run(self):
         self.axon.start()
 
         # --- Run until should_exit = True.
-        self.last_epoch_block = ttl_get_block(self)
-        bt.logging.info(f"Miner starting at block: {self.last_epoch_block}")
+        bt.logging.info(f"Miner starting at block: {self.block}")
 
         # This loop maintains the miner's operations until intentionally stopped.
         bt.logging.info(f"Starting miner main loop")
-        step = 0
         try:
             while not self.should_exit:
-                start_epoch = time.time()
-
-                # --- Wait until next epoch.
-                current_block = ttl_get_block(self)
                 while (
-                    current_block - self.last_epoch_block
-                    < self.config.neuron.checkpoint_block_length
```

Comment:
idea here is to generalize the name of the epoch length config parameter. if we design the template to do everything on the epoch there are a lot less flags. developers can then intoduce their own synchronization args.